Custom parameters support + Hook conflict fix (#182)
BugDiver/gauge-ts: docs/index.md

```

> ### Custom Paramter Parsers

Reviewer: ```suggestion
### Custom Parameter Parsers
```

> * By default gauge-ts tries to convert the spec parameter to primitives (number, boolean), for table parameters gauge-ts will convert paramters

Reviewer: ```suggestion
* By default gauge-ts tries to convert the spec parameter to primitives (number, boolean), for table parameters gauge-ts will convert parameters
```

> to `Table` type. gauge-ts also provide a way to customize the paramter parsing.

Reviewer: ```suggestion
to `Table` type. gauge-ts also provide a way to customize the parameter parsing.
```

> If you need to have custom paramters in your step implementations create a paramter parser which should implements methods from `ParamterParser` interface.

Reviewer: Should this be 
```suggestion
If you need to have custom paramters in your step implementations create a parameter parser which should implements methods from `ParameterParser` interface.
```

?

Step definition:
```md

* step with a person parameter type "{\"name\": \"John\", \"age\": 40 }"

```

Step Implementation:
```javascript
import { Step } from 'gauge-ts';

export class Person {
  name: string;
  age: number;
  constructor(name: string, age: number) {
    this.name = name;
    this.age = age;
  }

  public isAdult(): boolean {
    return this.age >= 18;
  }
}

export default class Implementation {
    @Step
    public async isAdult(person: Person)y {
      assert.ok(person.isAdult())
    }
}

```

> Cusotm ParameterParser:

Reviewer: ```suggestion
Custom ParameterParser:
```
